perun-registrar-lib/src/test/java/cz/metacentrum/perun/registrar/ExpirationNotifSchedulerTest.java

```diff
@@ -683,6 +686,66 @@ public void isCesnetEligible_ignoredCases() throws Exception {
 		isCesnetEligibleNotLogged();
 	}
 
+	// Test when a user is set to EXPIRED in VO, all of his group applications are rejected
+	@Test
+	public void expireMemberInVoRejectsGroupApplications() throws Exception {
+		System.out.println(CLASS_NAME + "expireMemberInVoRejectsGroupApplications");
+
+		// Create VO
+		Vo vo = new Vo(0, "testVo", "testVo");
+		vo = perun.getVosManagerBl().createVo(session, vo);
+
+		// Create Group
+		Group group = new Group("testGroup", "testGroup");
+		group = perun.getGroupsManagerBl().createGroup(session, vo, group);
+
+		// Create Member
+		User user = new User();
+		user.setFirstName("testFirstName");
+		user.setLastName("testLastName");
+
+		user = perun.getUsersManagerBl().createUser(session, user);
+		Member member = perun.getMembersManagerBl().createMember(session, vo, user);
+
+		// Create Group application form
+		registrarManager.createApplicationFormInGroup(session, group);
+
+		// Create application
+		Application application = new Application();
+		application.setExtSourceName("testExtSource");
+		application.setExtSourceType("testExtSourceType");
+		application.setExtSourceLoa(0);
+		application.setType(Application.AppType.INITIAL);
+		application.setGroup(group);
+		application.setUser(user);
+		application.setVo(vo);
+		application.setCreatedBy("testUser");
+		application.setCreatedAt(LocalDateTime.now().toString());
+
+		// Create form data
+		List<ApplicationFormItemData> data = new ArrayList<>();
+		ApplicationFormItemData itemData = new ApplicationFormItemData();
+		itemData.setShortname("email");
+		itemData.setValue("test@example.com");
+
+		registrarManager.submitApplication(session, application, data);
+
+		// Approve all applications
+		List<Application> applications = registrarManager.getApplicationsForGroup(session, group, new ArrayList<>());
+		for (Application app: applications) {
+			registrarManager.approveApplication(session, app.getId());
+		}
+
+		// Expire member
+		perun.getMembersManagerBl().setStatus(session, member, Status.EXPIRED);
```

Comment:
You will have to set the expiration date with you current implementation for this to work